### PR TITLE
Initial RKE cli engine deployment

### PR DESCRIPTION
**Pull request by bgentry-ntap**
-rke up cmd, no rancher server integration

### CODE REVIEW

**Comment by bgentry-ntap** [comment]
Lots of assumptions about the target nodes, rke installed, local host priv key can ssh via ssh user in config to all nodes, ssh user in docker user group. Tested with 1 control, 2 worker nodes lmk if you hit issues.

File: pkg/engines/rke-cli/rke-cli.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+func (c *MgmtCluster) CreatePermanent() error {
+	c.EventStream <- events.Event{EventType: "progress", Event: "install HA rke cluster"}
+
+	var y map[string]interface{}
+	err := yaml.Unmarshal([]byte(rawClusterYML), &y)
+	if err != nil {
+		return err
+	}
+
+	nodeKeys := make([]string, 0)
```
Comment (bgentry-ntap) [inline pkg/engines/rke-cli/rke-cli.go:77]: whoops this isn't used anymore. I will remove with other comments

File: pkg/engines/rke-cli/rke-cli.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	for k := range c.Nodes {
+		nodeKeys = append(nodeKeys, k)
+	}
+
+	nodes := make([]*rkeConfigNode, 0)
+	for k, v := range c.Nodes {
```
Comment (bgentry-ntap) [inline pkg/engines/rke-cli/rke-cli.go:83]: This does not support one node deployments, I can special case that to include all roles if we want.

File: cmd/deploy.go [hunk 3]
```diff
@@ -183,17 +184,32 @@ func runEngine() {
 		engineName = engine
 
 	} else if deploymentType == "rke" {
-		engine := rke.NewMgmtClusterFullConfig()
-		errJ := viper.Unmarshal(&engine)
-		if errJ != nil {
-			log.Fatalf("unable to decode into struct, %v", errJ.Error())
+		rkeDockerEnv := os.Getenv("CAKE_RKE_DOCKER")
```
Comment (jacobweinstock) [inline cmd/deploy.go:187]: is this to differentiate the engine from a local run vs on a vm we create? isnt that the idea of the local flag?

File: pkg/engines/rke-cli/rke-cli.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+	}
+	err = cmd.Start()
+	if err != nil {
+		return err
+	}
+	r := bufio.NewReader(stdout)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+	go func(ctx context.Context) {
```
Comment (jacobweinstock) [inline pkg/engines/rke-cli/rke-cli.go:137]: we should revisit this. once the command completes we could just check the exit status for success or failure. also we could just log the output to a file, that way a progress endpoint could stream that info.

File: pkg/engines/rke-cli/rke-cli.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+		}
+		if strings.HasPrefix(k, "controlplane") {
+			node.Role = append(node.Role, "controlplane")
+		} else {
+			node.Role = append(node.Role, "worker")
+		}
+		nodes = append(nodes, node)
+	}
+
+	// etcd requires an odd number of nodes, first role on each node is etcd.
```
Comment (jacobweinstock) [inline pkg/engines/rke-cli/rke-cli.go:106]: TODO: we're gonna need to cap the number control plane nodes to max at like 9. i think thats the largest recommended etcd cluster size for k8s

File: pkg/engines/rke-cli/rke-cli.go (hunk 2 above)
Comment (jacobweinstock) [reply pkg/engines/rke-cli/rke-cli.go:78]: for the first out i think this is fine.

File: cmd/deploy.go (hunk 3 above)
Comment (bgentry-ntap) [reply cmd/deploy.go:187]: No this is to differentiate RKE deployed from the docker container vs rke up on existing VM. Clearly not a good name :) Lmk if you have a better suggestion. I'll add a comment for now.

File: pkg/engines/rke-cli/rke-cli.go (hunk 4 above)
Comment (bgentry-ntap) [reply pkg/engines/rke-cli/rke-cli.go:137]: Yeah I still check the cmd status, I just also wanted the output streamed. I like the idea of streaming this to the progress endpoint

**Comment by bgentry-ntap** [comment]
> looks good for a first out. this is working in your manual testing, right?

Nah, testing is for production :P

File: cmd/deploy.go (hunk 3 above)
Comment (jacobweinstock) [reply cmd/deploy.go:189]: oh ok, i didn't realize you're not touching the existing stuff. looks like defaulting to rke up then? maybe just a short comment?

**Pull request merged**